pkg/controller/velerobackup/velerobackup_controller.go

```diff
@@ -82,6 +87,20 @@ func AddToManager(mgr manager.Manager, r reconcile.Reconciler) error {
 	}
 
 	reconciler := r.(*ReconcileBackup)
+
+	maxBackupFrequencySecondsStr := os.Getenv(hiveconstants.MaxBackupFrequencySecondsEnvVar)
```

Comment:
This is the first I'm hearing about your preference for having environment variables read and validated by the orchestrator and passed into the controller. Btw, I'm assuming by "orchestrator" you mean the hive "manager".

I did it this way because I was following the pattern established in our other controllers. I'm fine changing this, but I'd want to change it in all of our controllers so that we have 1 pattern to follow going forward. I'd also prefer that to be a separate card / PR. Let's discuss in the next stand up and if we decide to make the change overall, I'll add a card for it.

I also think that we need to come up with the best way to pass these into the controllers since the controller registration happens in package init() functions. So it gets tricky since that runs before the manager code. But that's all stuff we can work out. For instance, we don't necessarily need to use the package init() functions for registration.

In the mean time, I'll look at making the first change you requested.

Reply:
Oh, I did not mean to give the impression that I expected anything done about changing the (anti-)pattern in this PR. I was just grousing. We had a brief discussion of this same thing in standup a few weeks ago in relation to the clusterdeployment controller reading the global pull secret name from an environment variable.

Reply:
The "orchestrator" to which I am referring is the entity that creates all of the objects and wires them together. The controller creation does not happen in the `init` phase. It happens [here](https://github.com/openshift/hive/blob/30a1011410d191c011ebba314c60c1b2ede677b7/cmd/manager/main.go#L124). That object validates command-line arguments. It is reasonably to expect that it would also validate environment variables, since both of those are user inputs into the program.

Reply:
Seems reasonable for another PR when someone feels up for it.

Reply:
It would make the code feel a little more reusable and less magic.

Reply:
@staebler @dgoodwin I created [this](https://jira.coreos.com/browse/CO-581) card to move reading / validating environment variables from the individual controllers into hive manager.